### PR TITLE
add deployOnOCP field to the charts to support deploy on aks

### PR DESCRIPTION
**Pull request by zhiweiyin318**
# Description

add deployOnOCP field to the charts to support deploy on aks.

## Related Issue

https://issues.redhat.com/browse/ACM-11691

## Changes Made

Provide a clear and concise overview of the changes made in this pull request.

## Screenshots (if applicable)

Add screenshots or GIFs that demonstrate the changes visually, if relevant.

## Checklist

- [ ] I have tested the changes locally and they are functioning as expected.
- [ ] I have updated the documentation (if necessary) to reflect the changes.
- [ ] I have added/updated relevant unit tests (if applicable).
- [ ] I have ensured that my code follows the project's coding standards.
- [ ] I have checked for any potential security issues and addressed them.
- [ ] I have added necessary comments to the code, especially in complex or unclear sections.
- [ ] I have rebased my branch on top of the latest main/master branch.

## Additional Notes

Add any additional notes, context, or information that might be helpful for reviewers.

## Reviewers

Tag the appropriate reviewers who should review this pull request. To add reviewers, please add the following line: `/cc @reviewer1 @reviewer2`

## Definition of Done

- [ ] Code is reviewed.
- [ ] Code is tested.
- [ ] Documentation is updated.
- [ ] All checks and tests pass.
- [ ] Approved by at least one reviewer.
- [ ] Merged into the main/master branch.

### CODE REVIEW

File: pkg/rendering/renderer.go [hunk 1]
```diff
@@ -321,6 +329,9 @@ func injectValuesOverrides(values *Values, mch *v1.MultiClusterHub, images map[s
 
 	values.HubConfig.ClusterSTSEnabled = isSTSEnabled
 
+	// TODO: check if the operator is running on OCP cluster
+	values.Global.DeployOnOCP = "true"
```
Comment (mprahl) [inline pkg/rendering/renderer.go:333]: Is there a reason why this is a string and not a boolean?

**Comment by mprahl** [comment]
The GRC chart changes look okay to me!

File: pkg/rendering/renderer.go (hunk 1 above)
Comment (zhiweiyin318) [reply pkg/rendering/renderer.go]: nope.  boolean is better than string here.
changed to boolean.

**Comment by zhiweiyin318** [comment]
Thanks @mprahl  for reviewing.
btw, I think metric service is not needed in grc for aks case, but kube-rbac-proxy need the serving-cert of metric .  
can we remove the tls flags in the kube-rbac-proxy container?

**Comment by mprahl** [comment]
> Thanks @mprahl for reviewing. btw, I think metric service is not needed in grc for aks case, but kube-rbac-proxy need the serving-cert of metric . can we remove the tls flags in the kube-rbac-proxy container?

I think the metrics service is needed since the metrics it provides are used in the SD dashboard. I think kube-rbac-proxy likely doesn't need to protect the metrics endpoints with HTTPS since there is nothing sensitive exposed there. We may just need to document that limitation so that SD is aware.

**Pull request closed without merging**